NameLayerMain/src/vg/civcraft/mc/namelayer/permission/PermissionType.java

```diff
@@ -63,4 +37,44 @@ public static void displayPermissionTypes(Player p) {
 				+ "That PermissionType does not exists.\n"
 				+ "The current types are: " + getStringOfTypes());
 	}
+	
+	public static void registerPermission(PermissionType perm) {
+		permissions.put(perm.getName(), perm);
+	}
+	
+	public static Collection<PermissionType> values() {
+		return permissions.values();
+	}
+	
+	/**
+	 * 
+	 * @param name The name of the permissions. Not the PermissionFormat.
+	 * @return Returns the PermissionType.
+	 */
+	public static PermissionType getType(String name) {
+		return permissions.get(name);
+	}
+	
+	private String name;
+	private String permFormat;
+	/**
+	 * 
+	 * @param name The name of the permission.
+	 */
+	public PermissionType(String name) {
```

Comment:
Additionally I'd recommend that the constructor be private, with a public static `getInstance` be exposed; that way if the permission type already exists you return the one that lives in the map instead of potentially allocating a new one.